components/org.wso2.carbon.identity.tenant.resource.manager/src/main/java/org/wso2/carbon/identity/tenant/resource/manager/TenantAwareAxis2ConfigurationContextObserver.java

```diff
@@ -0,0 +1,188 @@
+        }
+    }
+
+    private List<EventPublisherConfiguration> getSuperTenantEventPublisherConfigurations(int tenantId) {
+
+        List<EventPublisherConfiguration> activeEventPublisherConfigurations = null;
+        try {
+            activeEventPublisherConfigurations = TenantResourceManagerDataHolder.getInstance()
+                    .getEventPublisherService().getAllActiveEventPublisherConfigurations();
+        } catch (EventPublisherConfigurationException e) {
+            log.error(handleServerException(
+                    TenantResourceConstants.ErrorMessages.ERROR_CODE_ERROR_WHEN_FETCHING_SUPER_TENANT_EVENT_PUBLISHER_CONFIGURATION,
+                    e, String.valueOf(tenantId)));
+        }
+        return activeEventPublisherConfigurations;
+    }
+
+    private List<EventStreamConfiguration> getSuperTenantEventStreamConfigurations(int tenantId) {
+
+        List<EventStreamConfiguration> eventStreamConfigurationList = null;
+        try {
+            eventStreamConfigurationList = TenantResourceManagerDataHolder.getInstance().getCarbonEventStreamService()
+                    .getAllEventStreamConfigurations();
+        } catch (EventStreamConfigurationException e) {
+            log.error(handleServerException(
+                    TenantResourceConstants.ErrorMessages.ERROR_CODE_ERROR_WHEN_FETCHING_SUPER_TENANT_EVENT_STREAM_CONFIGURATION,
+                    e, String.valueOf(tenantId)));
+        } finally {
+            PrivilegedCarbonContext.endTenantFlow();
+        }
+        return eventStreamConfigurationList;
+    }
+
+    private void startTenantFlow(int tenantId, String tenantDomain) {
+
+        PrivilegedCarbonContext.startTenantFlow();
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantId(tenantId);
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain);
+    }
+
+    private void startSuperTenantFlow() {
+
+        PrivilegedCarbonContext.startTenantFlow();
+        PrivilegedCarbonContext carbonContext = PrivilegedCarbonContext.getThreadLocalCarbonContext();
+        carbonContext.setTenantId(MultitenantConstants.SUPER_TENANT_ID);
+        carbonContext.setTenantDomain(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
+    }
+
+    private void createTenantEventPublisherConfiguration(int tenantId,
+            List<EventPublisherConfiguration> activeEventPublisherConfigurations) {
+
+        for (EventPublisherConfiguration eventPublisherConfiguration : activeEventPublisherConfigurations) {
+            try {
+                ResourceFile resourceFile = getResourceFile(eventPublisherConfiguration.getEventPublisherName());
+                if (resourceFile != null) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Configuration store is enabled and there is a file in the configuration store for "
+                                + "the publisher name: " + eventPublisherConfiguration.getEventPublisherName());
```

Comment:
Does 'configuration store is enabled' make sense here.
It's something related to the configuration store.
Moreover, if it's not enabled doesn't it throw some exception. If so you can this at such point. If not this consumer is not aware if it's enabled or not. It just knows whether a resource was found or not

Reply:
changed